sqlmesh/utils/pandas.py

```diff
@@ -34,7 +34,11 @@
 def columns_to_types_from_df(df: pd.DataFrame) -> t.Dict[str, exp.DataType]:
     result = {}
     for column_name, column_type in df.dtypes.items():
-        exp_type = PANDAS_TYPE_MAPPINGS.get(column_type)
+        exp_type: t.Optional[exp.DataType] = None
+        if hasattr(pd, "DatetimeTZDtype") and isinstance(column_type, pd.DatetimeTZDtype):
+            exp_type = exp.DataType.build("timestamp")
```

Comment:
Why are we constructing a `Type.TIMESTAMP` instead of `Type.TIMESTAMPTZ`, given that the `column_type` is a timezone-aware datetime? Also, should we be `build`ing this `DataType` in a dialect-agnostic way?

Reply:
This code was written before `DataType.build` required the dialect. This change is not actually necessary for my PR. Let me revert this and address this separately.

Reply:
Actually, it looks like the dialect is only used to parse the input argument. Provided my values are hardcoded and correspond to the default dialect anyway I don't think I need an explicit dialect in here. I will switch to using TIMESTAMPZ

Reply:
Fixed this